src/sentry/tagstore/snuba/backend.py

```diff
@@ -440,7 +440,9 @@ def get_tag_keys_for_projects(
         # the sampling that turbo enables so that we get more accurate results.
         # We only want sampling when we have a large number of projects, so
         # that we don't cause performance issues for Snuba.
-        if len(projects) <= max_unsampled_projects:
+        # We also see issues with long timeranges in large projects,
+        # So only disable sampling if the timerange is short enough.
+        if len(projects) <= max_unsampled_projects and end - start > timedelta(days=14):
             optimize_kwargs["sample"] = 1
```

Comment:
What is the sample rate that this eventually sets in the request?

Reply:
looks like 10%? https://github.com/getsentry/snuba/blob/b88683fae94b6a3835b4db10793c7c6ccc0fb723/snuba/settings/__init__.py#L267